internal/app/machined/pkg/controllers/network/dns_resolve_cache.go

```diff
@@ -288,3 +295,29 @@ func newDNSRunner(cfg runnerConfig, cache *dns.Cache, logger *zap.Logger) (*dns.
 
 	return dns.NewServer(serverOpts), nil
 }
+
+type stateMapper struct {
+	state state.State
+}
+
+func (s *stateMapper) ResolveAddr(ctx context.Context, name string) (netip.Addr, bool) {
+	name = strings.TrimRight(name, ".")
+
+	list, err := safe.ReaderListAll[*cluster.Member](ctx, s.state)
+	if err != nil {
+		return netip.Addr{}, false
+	}
+
+	elem, ok := list.Find(func(res *cluster.Member) bool { return res.TypedSpec().Hostname == name })
```

Comment:
I'm not sure that `id == .Hostname` is always true so I went with this. If it is always true then this can be dropped and we can get spec by id above.

Reply:
as we discussed, `Hostname` might be a FQDN as well, so the match should be updated. 

If we match on the first part of the FQDN, it's a match (`foo` -> `foo.example.com`).
If we match on the full name, it's a match (`foo.example.com` -> `foo.example.com`).
We should also probably match on `Nodename` same way as well.

Reply:
Done. We are also testing this matching in controller unit-tests.